### PR TITLE
Weekly Patch Release v1.5.9

### PR DESCRIPTION
**Pull request by justusschock**
## What does this PR do?

Contains all bugfixes to be released next week. As bugfixes get merged to master, they will get cherry-picked into this branch by the individual author (core) or on-caller.

For staff/core: The new [release guide is here](https://www.notion.so/gridai/Release-Guide-ab132bf94f304056afc8ff992304c18f).

```
$ gh pr list -s merged -S 'merged:2022-01-06T18:30:00.000Z..2022-01-18T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "1.5.x") or (.milestone.title == null)) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100

2022-01-12T08:09:36Z https://github.com/PyTorchLightning/pytorch-lightning/pull/11401 f5bbc2cf174a9942f4c2dc22fc3735aa7e50dd93 Avoid in-place ops during logging result updates
2022-01-10T23:12:53Z https://github.com/PyTorchLightning/pytorch-lightning/pull/11400 34c62da37dc1ed9a1de7e023c690c7528ee56c60 Pin sphinx-autodoc-typehints with <v1.15
2022-01-07T13:09:53Z https://github.com/PyTorchLightning/pytorch-lightning/pull/11217 8e8d4c5fe02467374b51c3220dfa70dfeb8a9234 Skip testing with PyTorch 1.7 and Python 3.9 on Ubuntu
```

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda

### CODE REVIEW

**Comment by rohitgr7** [comment]
do we need to squash the extra commits? @justusschock

**Comment by kaushikb11** [comment]
@justusschock No changelog updates?

**Comment by justusschock** [comment]
@rohitgr7 probably not necessary if we do a squash merge 

@kaushikb11 oops, just forgot to push. Will add them later when back on pc :)

**Comment by rohitgr7** [comment]
> @rohitgr7 probably not necessary if we do a squash merge
> 
> @kaushikb11 oops, just forgot to push. Will add them later when back on pc :)

for weekly releases, we do `Rebase + Merge` to keep the cherry-picked commits

**Comment by akihironitta** [comment]
Converting to draft for this PR to be released next week.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by akihironitta — APPROVED** [review]

**Comment by carmocca** [comment]
Included #11366 because we are getting this error on some runs testing "oldest" versions

```python
INTERNALERROR>   File "C:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\pytest_asyncio\plugin.py", line 191, in pytest_configure
INTERNALERROR>     config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
INTERNALERROR> AttributeError: 'Config' object has no attribute 'issue_config_time_warning'
```

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ All notable changes to this project will be documented in this file.
...
+
+- Pin sphinx-autodoc-typehints with <v1.15 ([#11400](https://github.com/PyTorchLightning/pytorch-lightning/pull/11400))
+- Skip testing with PyTorch 1.7 and Python 3.9 on Ubuntu ([#11217](https://github.com/PyTorchLightning/pytorch-lightning/pull/11217))
+- Fixed type promotion when tensors of higher category than float are logged ([#11401](https://github.com/PyTorchLightning/pytorch-lightning/pull/11401))
+- Fixed the format of the configuration saved automatically by the CLI's `SaveConfigCallback` ([#11532](https://github.com/PyTorchLightning/pytorch-lightning/pull/11532))
+
+### Changed
+
+- Changed `LSFEnvironment` to use `LSB_DJOB_RANKFILE` environment variable instead of `LSB_HOSTS` for determining node rank and main address ([#10825](https://github.com/PyTorchLightning/pytorch-lightning/pull/10825))
+- Disbled sampler replacement when using `IterableDataset` ([#11507](https://github.com/PyTorchLightning/pytorch-lightning/pull/11507))
```
Comment (ananthsub) [inline CHANGELOG.md:19]:
```suggestion
- Disabled sampler replacement when using `IterableDataset` ([#11507](https://github.com/PyTorchLightning/pytorch-lightning/pull/11507))
```

**Pull request merged**